### PR TITLE
Do not include a trailing new line in the ABI JSON output

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
@bobsummerwill @chriseth Seems like the library feature I'm using is present since about the [0.7.0 release](https://github.com/open-source-parsers/jsoncpp/releases/tag/0.7.0) of jsoncpp.

Travis on Linux is using this outdated version from 2011:

```
Setting up libjsoncpp-dev:amd64 (0.6.0~rc2-3ubuntu1) ...
```

The latest version is 1.7.4. Since [version 0.9.4](https://github.com/open-source-parsers/jsoncpp/releases/tag/0.9.4) has "full UTF-8 support", maybe it would make sense recommending that version and having it available in Travis?

Travis on OSX has a newer version which compiles.

**Comment by bobsummerwill** [comment]
Right.   And we're using a newer version for Windows too:   https://github.com/ethereum/cpp-dependencies/releases/tag/vc140

Is there a simple "if jsoncpp version > x.x.x" conditional which you could put around this call as a stop-gap measure?

Then I can also update the `install_deps.sh` for Solidity as part of my pending "Solidity standalone" PR, where we would do an install-from-source to get a newer version, and could remove the conditional again at that point.

**Comment by bobsummerwill** [comment]
When Solidity standalone and cpp-ethereum-v1.3.1 "land", both the Travis/Appveyor and "building from source" instructions will switch to use `install_deps.sh`, where we still have manual per-distro instructions at the moment.

**Comment by chriseth** [comment]
Please rebase.

File: libsolidity/interface/InterfaceHandler.cpp [hunk 1]
```diff
@@ -95,7 +95,16 @@ string InterfaceHandler::abiInterface(ContractDefinition const& _contractDef)
...
-	return Json::FastWriter().write(abi);
+
+	Json::FastWriter writer;
+#if JSONCPP_VERSION_HEXA >= 0x700
+	writer.omitEndingLineFeed();
+	return writer.write(abi);
+#else
+	// Remove trailing new line
+	string ret = writer.write(abi);
+	return !ret.empty() ? ret.erase(ret.length() - 1) : ret;
```
Comment (chriseth) [inline libsolidity/interface/InterfaceHandler.cpp]: Please check that it actually is a newline - just in case.

**Comment by axic** [comment]
Perhaps better to wait for the outcome of #789. That could simplify the above code.

**Comment by axic** [comment]
Can be merged after #1252.

**Comment by axic** [comment]
Actually this made me wondering. Every JSON output (ie. libevmasm) is passed as a `JsonValue`, except the interface, which outputs `string`.

It perhaps would make sense changing that to `JsonValue` too and keep every JSON serialisation call within the `jsonCompiler`.

**Comment by chriseth** [comment]
Yes, makes sense. I think in the beginning we were reluctant to spill json across the codebase.

File: Changelog.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 Bugfixes:
  * Type checker: forbid signed exponential that led to an incorrect use of EXP opcode.
  * Code generator: properly clean higher order bytes before storing in storage.
+ * Interface JSON: do not include trailing new line.
```
Comment (chriseth) [inline Changelog.md]: Please rebase and move to correct verison.

Comment (axic) [reply Changelog.md]: @chriseth done

**Review by chriseth — APPROVED** [review]

**Pull request merged**